### PR TITLE
Updated Version.java path from server to libs in version.yml

### PR DESCRIPTION
**Pull request by vibrantvarun**
### Description
Updating Version.java path in version.yml from server/src/main/java/org/opensearch/Version.java to libs/core/src/main/java/org/opensearch/Version.java

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kotwanikunal** [comment]
@vibrantvarun Can you please describe what you are trying to achieve in the description section of the PR?

**Comment by prudhvigodithi** [comment]
Thanks @vibrantvarun I assume the `sed` logic still works with the file `buildSrc/src/main/java/org/opensearch/gradle/Version.java`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Exclude 'benchmarks' from codecov report ([#8805](https://github.com/opensearch-project/OpenSearch/pull/8805))
 - [Refactor] MediaTypeParser to MediaTypeParserRegistry ([#8636](https://github.com/opensearch-project/OpenSearch/pull/8636))
 - Create separate SourceLookup instance per segment slice in SignificantTextAggregatorFactory ([#8807](https://github.com/opensearch-project/OpenSearch/pull/8807))
+- Updated Version.java path from server to buildSrc in version.yml ([#8883](https://github.com/opensearch-project/OpenSearch/pull/8883))
```
Comment (andrross) [inline CHANGELOG.md:99]: Build/test related changes can be omitted from the changelog

File: .github/workflows/version.yml [hunk 2]
```diff
@@ -54,8 +54,8 @@ jobs:
           echo "  - \"$CURRENT_VERSION\"" >> .ci/bwcVersions
           sed -i "s/opensearch        = $CURRENT_VERSION/opensearch        = $NEXT_VERSION/g" buildSrc/version.properties
           echo Adding $NEXT_VERSION_UNDERSCORE after $CURRENT_VERSION_UNDERSCORE
-          sed -i "s/public static final Version $CURRENT_VERSION_UNDERSCORE = new Version(\([[:digit:]]\+\)\(.*\));/\0\n    public static final Version $NEXT_VERSION_UNDERSCORE = new Version($NEXT_VERSION_ID\2);/g" server/src/main/java/org/opensearch/Version.java
-          sed -i "s/CURRENT = $CURRENT_VERSION_UNDERSCORE;/CURRENT = $NEXT_VERSION_UNDERSCORE;/g" server/src/main/java/org/opensearch/Version.java
+          sed -i "s/public static final Version $CURRENT_VERSION_UNDERSCORE = new Version(\([[:digit:]]\+\)\(.*\));/\0\n    public static final Version $NEXT_VERSION_UNDERSCORE = new Version($NEXT_VERSION_ID\2);/g" buildSrc/src/main/java/org/opensearch/gradle/Version.java
```
Comment (andrross) [inline .github/workflows/version.yml:57]: `libs/core/src/main/java/org/opensearch/Version.java` is the Version file you're looking for.

File: CHANGELOG.md (hunk 1 above)
Comment (vibrantvarun) [reply CHANGELOG.md:99]: Build is failing if changelog is not added then

Comment (andrross) [reply CHANGELOG.md]: I've added the skip-changelog label

**Comment by vibrantvarun** [comment]
@andrross sure, taking a look into it.

File: CHANGELOG.md (hunk 1 above)
Comment (vibrantvarun) [reply CHANGELOG.md]: ACK

**Review by scrawfor99 — APPROVED** [review]

**Comment by owaiskazi19** [comment]
> @vibrantvarun Can you please describe what you are trying to achieve in the description section of the PR?

Looks like these changes were missed [here](https://github.com/opensearch-project/OpenSearch/pull/7328/files#diff-fd17c77905dbf19e585660488bae21de251453d5389f9173b2a0138cfb16e3a9L108).

**Comment by kotwanikunal** [comment]
> > @vibrantvarun Can you please describe what you are trying to achieve in the description section of the PR?
> 
> Looks like these changes were missed [here](https://github.com/opensearch-project/OpenSearch/pull/7328/files#diff-fd17c77905dbf19e585660488bae21de251453d5389f9173b2a0138cfb16e3a9L108).

Thanks. I was looking for this context :)

**Comment by kotwanikunal** [comment]
> > Thanks @vibrantvarun I assume the `sed` logic still works with the file `buildSrc/src/main/java/org/opensearch/gradle/Version.java`.
> 
> @vibrantvarun Can you run the `sed` commands manually against the correct file and verify it still works? @kotwanikunal please chime in if there's a better way to test this.

That's the only way. @vibrantvarun You can test this by cloning OpenSearch and running this script from the root folder - 
https://gist.github.com/kotwanikunal/91f33eeced50f6aa93701e91a2e93a6b

**Comment by vibrantvarun** [comment]
@kotwanikunal 
The output of running above script on local
```
TAG=2.9.0
BASE=2.9
BASE_X=2.x
CURRENT_VERSION=2.9.0
CURRENT_VERSION_UNDERSCORE=V_2_9_0
NEXT_VERSION=2.9.1
NEXT_VERSION_UNDERSCORE=V_2_9_1
NEXT_VERSION_ID=2090199
Incrementing 2.9.0 to 2.9.1
version.sh: line 32: .ci/bwcVersions: No such file or directory
sed: 1: "buildSrc/version.proper ...": undefined label 'uildSrc/version.properties'
Adding V_2_9_1 after V_2_9_0
```

It works.

**Comment by kotwanikunal** [comment]
> @kotwanikunal The output of running above script on local
> 
> ```
> TAG=2.9.0
> BASE=2.9
> BASE_X=2.x
> CURRENT_VERSION=2.9.0
> CURRENT_VERSION_UNDERSCORE=V_2_9_0
> NEXT_VERSION=2.9.1
> NEXT_VERSION_UNDERSCORE=V_2_9_1
> NEXT_VERSION_ID=2090199
> Incrementing 2.9.0 to 2.9.1
> version.sh: line 32: .ci/bwcVersions: No such file or directory
> sed: 1: "buildSrc/version.proper ...": undefined label 'uildSrc/version.properties'
> Adding V_2_9_1 after V_2_9_0
> ```
> 
> It works.

I can see errors on the output. Did you test it against the OpenSearch repo cloned code base?

**Comment by vibrantvarun** [comment]
@kotwanikunal Okay I debugged the issue with the script. Actually in mac os the sed command looks like

sed -i -e "s/opensearch        = $CURRENT_VERSION/opensearch        = $NEXT_VERSION/g" buildSrc/version.properties

We have to add **-e** in the command

So after I added -e below is the output
```
TAG=2.9.0
BASE=2.9
BASE_X=2.x
CURRENT_VERSION=2.9.0
CURRENT_VERSION_UNDERSCORE=V_2_9_0
NEXT_VERSION=2.9.1
NEXT_VERSION_UNDERSCORE=V_2_9_1
NEXT_VERSION_ID=2090199
Incrementing 2.9.0 to 2.9.1
Adding V_2_9_1 after V_2_9_0
```

**Review by kotwanikunal — APPROVED** [review]

**Comment by kotwanikunal** [comment]
@vibrantvarun Please resolve all conversations before we can merge this in.

**Comment by vibrantvarun** [comment]
@kotwanikunal resolved all conversations

**Review by ryanbogan — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Hey @vibrantvarun this only runs when a new tag is cut which takes in the tag information https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/version.yml#L24. So to re-produce one way is to delete and re-create tag which is no right thing to do now, please raise a manual PR for `2.9.1` similar to [2.8.1](https://github.com/opensearch-project/OpenSearch/commit/2c3366916940a90c91da59e8d981b810659a2468) and lets keep monitoring for the next patch version increment.
Thank you

**Pull request merged**